torch/distributed/_tensor/dispatch.py

```diff
@@ -216,17 +226,19 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
                         f"return type {ret_type} in DTensor op is not supported"
                     )
     else:
-        # compute locally with redistribute first if needed
-        local_tensor_args = pack_args_kwargs_with_local_tensor(
-            args,
-            suggested_input_schema.args_schema,
-            redistribute_with_schema=needs_redistribute,
+        if output_sharding.needs_redistribute:
+            # compute locally with redistribute first if needed
+            assert output_sharding.schema_suggestions is not None
+            suggested_input_schema = output_sharding.schema_suggestions[0]
+            redistribute_local_args(op_info, suggested_input_schema)
```

Comment:
This name is a little bit confusing to me because the concept `local` to me is more like tensor rather DTensor while redistribute is a unique thing for DTensor. Would it be possible that we can just call it `redistribute_args` or `redistribute_n_update_local_args`? I understand the logic here is to first redistribute dtensor and update local_args so that we pass the correct args to the final Aten ops. But the naming seems to suggest we are doing redistribute directly on top of a Tensor.

Reply:
After reading your comment below and a second thought... I think this name is fine.